defog/__init__.py

```diff
@@ -227,22 +227,30 @@ def generate_postgres_schema(self, tables: list, upload: bool = True) -> str:
             cur.execute(
                 "SELECT table_name FROM information_schema.tables WHERE table_schema = 'public';"
             )
-            tables = [row[0] for row in cur.fetchall()]
+            tables = [f"public.{row[0]}" for row in cur.fetchall()]
+        else:
+            for table in tables:
+                if not table or len(table.split("."))!=2:
+                    raise ValueError(f"PostgreSQL table names should be of the following format <schema>.<table> which is violated by '{table}`")
+
         print("Retrieved the following tables:")
+
         for t in tables:
             print(f"\t{t}")
 
         print("Getting schema for each table in your database...")
         # get the schema for each table
-        for table_name in tables:
+        for table in tables:
+            
+            schema,table_name = table.split(".")
```

Comment:
To avoid needing to throw an exception, one suggestion is to first check if there is a `.`, and if there isn't, to default the schema to `public`. eg:
```python
for table in tables:
  if '.' in table:
    schema, table_name = table.split(".",1)
  else:
   schema, table_name = "public", table
```